components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1744,12 +1745,27 @@ public IdentityProvider addIdPWithResourceId(IdentityProvider identityProvider,
     @Override
     public void deleteIdP(String idPName, String tenantDomain) throws IdentityProviderManagementException {
 
+        // Invoking the pre listeners.
+        Collection<IdentityProviderMgtListener> listeners = IdPManagementServiceComponent.getIdpMgtListeners();
+        for (IdentityProviderMgtListener listener : listeners) {
+            if (listener.isEnable() && !listener.doPreDeleteIdP(idPName, tenantDomain)) {
+                return;
+            }
+        }
+
         IdentityProvider identityProvider = this.getIdPByName(idPName, tenantDomain, true);
         if (identityProvider == null) {
             throw IdPManagementUtil.handleClientException(IdPManagementConstants.ErrorMessage
-                    .ERROR_CODE_IDP_DOES_NOT_EXIST, idPName);
+                    .ERROR_CODE_IDP_NAME_DOES_NOT_EXIST, idPName);
```

Comment:
If IDP does not exist, we should simply return since the purpose of delete operation has already completed if the resource does not exists.